### PR TITLE
Air Alarms are now visible in the dark

### PR DESCRIPTION
**Pull request by Poojawa**
:cl: Poojawa
add: Air Alarms are now visible in poor lighting conditions
/:cl:

I dunno about you, but seeing red APCs in the dark is pretty good to know. But if I'm peeking out into the darkness and trying to see if the hallway is vented, why...

That's not very helpful at all. But y'know what _would_ be?

Hopefully not...

But this is probably okay, enough.

I figured I'd spruce up Air alarms to use fancy `SSvis_overlays` like Computers and APCs do now. There's a very soft glow to each, despite the screenshots looking terribly dark. These are pretty coder sprite potato quality,  but usually people understand what the colors mean.

https://www.dropbox.com/s/3qegrmweks5sep6/airalarm2.mp4?dl=0 here's a video of how they look. Despite everything there's still a groddamn lack of seeing the glow, but I swears it's there. It's as powerful as the arcade machine's

### CODE REVIEW

**Comment by Cyberboss** [comment]
I like what you've done with this, but the highlight seems a little janky

**Comment by Poojawa** [comment]
Indeed. The only one that came out decently was the red, the others are still a bit weird, but I'll try fluffing out the highlights a little

**Comment by Poojawa** [comment]
amber and red light
amber and red dark
green light
green dark

How say you to these?

**Comment by Poojawa** [comment]
Updooted with revised and tweaked sprite states. Namely giving the 'off' tick a darker, non-highlight state so they're more legible in bright lighting.

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 1]
```diff
@@ -212,7 +217,11 @@
 
 /obj/machinery/airalarm/Initialize(mapload)
 	. = ..()
+	if(!mapload)
+		return
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: What's this block for? Why should this not run the other procs if instanced in regular play?

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 2]
```diff
@@ -198,12 +204,11 @@
 		panel_open = TRUE
 		pixel_x = (dir & 3)? 0 : (dir == 4 ? -24 : 24)
 		pixel_y = (dir & 3)? (dir == 1 ? -24 : 24) : 0
+		update_icon()
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: You shouldn't need this. `update_icon()` is called in `power_change()` which is called during init. Just remove this from new.

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 3]
```diff
@@ -198,12 +204,11 @@
...
 
 	if(name == initial(name))
 		name = "[get_area_name(src)] Air Alarm"
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: While you're dealing with the new proc here transfer the other easy to transfer things like this to init

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 4]
```diff
@@ -212,7 +217,11 @@
 
 /obj/machinery/airalarm/Initialize(mapload)
 	. = ..()
+	if(!mapload)
+		return
+	power_change()
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: This is already called in `/obj/machinery/Initialize()`

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 5]
```diff
@@ -212,7 +217,11 @@
...
 	set_frequency(frequency)
+	update_icon()
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: This is already called in `power_change()`

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by ChangelingRain — COMMENTED** [review]
The letters probably shouldn't light up; it makes the sprite too busy.
The glow around the state indicator should be a rectangle, instead of missing corners.
`alarm_green` also shouldn't flash or have a ! on it, since it's the neutral state.

My feedback boils down to "these don't quite fit the sprite and are slightly lower-quality than the current air alarms"

**Comment by ShizCalev** [comment]
#40048 +1

**Comment by Poojawa** [comment]
Like the console screens, I don't really know how to fix FoW stuff. I'm guessing if you put it on top of the lighting plane it borks *all* of it up.

**Comment by ninjanomnom** [comment]
It isn't fixable at all for us really without doing some really hacky shit. It needs engine features to handle properly. This is why I stopped making more of these.

**Comment by Iamgoofball** [comment]
@Poojawa get yourself shareX and use webm

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 6]
```diff
@@ -826,8 +856,12 @@
 	playsound(src, "sparks", 50, 1)
 
 /obj/machinery/airalarm/obj_break(damage_flag)
-	..()
-	update_icon()
+	if(circuit && !(flags_1 & NODECONSTRUCT_1)) //no circuit, no breaking
+		if(!(stat & BROKEN))
+			playsound(loc, 'sound/effects/glassbr3.ogg', 100, 1)
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/airalarm.dm]: condense the 2 ifs and change loc to src

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 7]
```diff
@@ -826,8 +856,11 @@
 	playsound(src, "sparks", 50, 1)
 
 /obj/machinery/airalarm/obj_break(damage_flag)
-	..()
-	update_icon()
+	if(circuit && !(flags_1 & NODECONSTRUCT_1) && !(stat & BROKEN))
+		playsound(src, 'sound/effects/glassbr3.ogg', 100, 1)
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/airalarm.dm]: Why'd you remove the ..() call?

Comment (Poojawa) [reply code/modules/atmospherics/machinery/airalarm.dm]: Oh, that might actually be a me-goof, lemmie just revert it

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 8]
```diff
@@ -816,7 +841,12 @@
 
 /obj/machinery/airalarm/power_change()
 	..()
+	if(stat & NOPOWER)
+		set_light(0)
+	else
+		set_light(brightness_on)
 	update_icon()
+	return
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: This return is unnecessary

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 9]
```diff
@@ -816,7 +841,12 @@
 
 /obj/machinery/airalarm/power_change()
 	..()
+	if(stat & NOPOWER)
+		set_light(0)
+	else
+		set_light(brightness_on)
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: update_icon() called just below also sets the light, you have no reason to do it here

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 10]
```diff
@@ -629,22 +645,29 @@
...
+			light_color = LIGHT_COLOR_RED
+			set_light(brightness_on)
+
+	SSvis_overlays.add_vis_overlay(src, icon, overlay_state, ABOVE_LIGHTING_LAYER, ABOVE_LIGHTING_PLANE, dir)
+	update_light()
 
 /obj/machinery/airalarm/process()
 	if((stat & (NOPOWER|BROKEN)) || shorted)
-		return
+		return FALSE
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: process doesn't have a return value it cares about except `PROCESS_KILL` so there's no reason to add one here

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 11]
```diff
@@ -681,6 +704,8 @@
 		mode = AALARM_MODE_SCRUBBING
 		apply_mode()
 
+	return TRUE
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: ^

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 12]
```diff
@@ -202,17 +209,17 @@
...
 	SSradio.remove_object(src, frequency)
 	qdel(wires)
 	wires = null
 	return ..()
 
-/obj/machinery/airalarm/Initialize(mapload)
-	. = ..()
-	set_frequency(frequency)
+/obj/machinery/airalarm/on_construction()
+	..(dir,dir)
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: `/machinery` doesn't have any code to run for this proc and doesn't take any args. What are you doing here?

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 13]
```diff
@@ -681,6 +701,8 @@
 		mode = AALARM_MODE_SCRUBBING
 		apply_mode()
 
+	return
+
```
Comment (ninjanomnom) [inline code/modules/atmospherics/machinery/airalarm.dm]: You dont need returns at the end of procs like this

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by ninjanomnom** [comment]
As I said to oranges earlier I stopped making more things overlaid on top of the lighting layer because it has unsolvable issues without changes in byond. I wouldn't mind it much but honestly we probably shouldn't be using this until we can make sure it doesn't overlay on top of things like megafauna.

**Comment by Poojawa** [comment]
I'd say that's really just a minor cosmetic issue. There's just really no other practical means other than making the air alarms super bright - which just would be ugly, imo.

**Comment by ninjanomnom** [comment]
image
The issue isnt with vis overlays, it's with the feature concept as a whole. How does adding extra overlays fix anything about that?

**Comment by Poojawa** [comment]
If you checked the comment, I explained why. I've noticed that sometimes the client doesn't have the vis_overlays actually overlaid, usually at roundstart upon instantly joining on box station etc, so air alarms would appear as just blank units. now they've got an overlay that'll show status, if not glow in the dark.

**Comment by ninjanomnom** [comment]
Don't bandaid fix that issue, it's a byond problem which needs to get fixed

**Comment by ninjanomnom** [comment]
You can't fix it, it's a byond problem

**Comment by Poojawa** [comment]
So should I close this or...?

**Comment by optimumtact** [comment]
stop using his bogus ass vis overlay thing for one then maybe it would work

**Comment by ninjanomnom** [comment]
Regular overlays have the same issue

**Comment by optimumtact** [comment]
for now lets not do this for anything else other than computer screens

**Comment by Poojawa** [comment]
I mean, I guess? It's been merged on Citadel for a few weeks now with no ill issues.

**Comment by optimumtact** [comment]
good for them, but this has layering issues with stuff appearing over the FOV layers

**Pull request closed without merging**